core/src/main/java/io/ably/lib/realtime/Presence.java

```diff
@@ -872,7 +881,7 @@ synchronized Collection<PresenceMessage> get(Param[] params) throws AblyExceptio
          * false if the message is already superseded
          */
         synchronized boolean put(PresenceMessage item) {
-            String key = item.memberKey();
+            String key = item.clientId; //RTP17h
```

Comment:
RTP17h says only the presence map containing members that match the current connectionId should be keyed by client ID - though this looks like both maps are being keyed by client ID? (both `PresenceMap presence` and `PresenceMap internalPresence`?)

Reply:
It seems spec are changed as I cannot find RTP17h in https://github.com/ably/docs/pull/1526/files

Reply:
Its moved to https://github.com/ably/specification/pull/88/files

Reply:
It looks like there is only one presence map in java as I cannot find another one.
Member key seems deprecated by RTP17h from this [internal slack message](https://ably-real-time.slack.com/archives/C02P7F81DPS/p1664904124309399) or I did not get this correctly?

Reply:
I mean there are maps `presence` and `internalPresence` right?
```
    private final PresenceMap presence = new PresenceMap();
    private final PresenceMap internalPresence = new PresenceMap();
```

My understanding from the spec was:
* The `presence` map is the main mapping from `memberKey`s to presence message - from spec RTP2
* The `internalPresence` only contains the members that match the current connectionId - mapping `clientId` to presence message - from spec RTP17h

So `PresenceMap` needs to be indexed differently depending on its type? eg from ably-js https://github.com/ably/ably-js/pull/1045/commits/a036b0d5ff892ab2a577afd73ce4b1ca260f75c9

Reply:
@andydunstall I have refactored it to match JS SDK, commit is here https://github.com/ably/ably-java/pull/842/commits/86625b85409b7a78960b9680e299733e09dec90b. Thank you for your input.